### PR TITLE
BUG: Fix incorrect Y2R when recovered year followed by lesser recovered values

### LINKED ISSUE #41: Unexpected Y2R values
In some cases, `RestorationArea.yr2()` is producing unexpected results. e.g saying 1 year for recovery when it should be 0. This behavior is not consistent for all inputs, and the current set of units tests are all passing, so something weird is going on. A more detailed description with examples of the behaviour will be added soon.

### PR DESCRIPTION
**Pull request by szwiep**
This PR includes an updated `metrics.y2r` method which correctly takes the first year to meet or exceed a recovery threshold in a multi-dimensional timeseries xarray.DataArray. Additionally, new unit tests are added to catch the previously missed corner case.

Previous implementations of `metrics.y2r` incorrectly used the [idxmin](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.idxmin.html) method to find the first recovered year. This meant that if a recovered year was followed by a value that is also recovered, but less than the recovered year, then the lesser value/year would be incorrectly returned. See https://github.com/PEOPLE-ER/Spectral-Recovery/issues/41#issuecomment-1883897760 for more info.

This PR implements a new method using [argmin](https://docs.xarray.dev/en/stable/generated/xarray.DataArray.argmax.html) that correctly returns the first recovered year regardless of successive values, as well as passes all existing unit tests.

Resolves #41

### CODE REVIEW

File: docs/overview_API.ipynb
Comment (KavlinClein) [inline docs/overview_API.ipynb:1]: In the Polygons section:
- Text line: TYPO - should be "Read _your_ restoration polygon"

In Input Parameters table: 
- Might be worth explaining how to use certain parameters when working with a reference method or a historical method? Like two small example cells showing how you might write the code for each use case? Or maybe just adding which parameters are optional could be helpful?
- Restoration Start & Disturbance Start cells
                 -Disturbance and reference descriptions are flipped.


Other than that it seems everything is running well and I think this is helpful. Thanks, for putting this together!

**Review by KavlinClein — CHANGES_REQUESTED** [review]
Thanks for putting this together! Just left a small comment regarding a typo in overview_API.ipynb as well as a couple of other small changes I suggest.

File: docs/overview_API.ipynb
Comment (szwiep) [reply docs/overview_API.ipynb:1]: > might be worth explaining how to use certain parameters when working with a reference method or a historical method?

I think this is a good band-aid fix to do until we do an overhaul on the docs.

Comment (szwiep) [reply docs/overview_API.ipynb:1]: @KavlinClein would you be OK if I do these doc updates on a new feature branch? Seems a bit far away from the Y2R bug to be included in the same PR.

**Pull request merged**